.github/workflows/testing-farm.yml

```diff
@@ -0,0 +1,28 @@
+name: Testing Farm test
+on:
+  pull_request: {}
+
+jobs:
+  tests:
+    runs-on: [self-hosted, linux, testing-farm]
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: prepare env
+        run: |
+          sudo yum install podman openssh-server -y
+          sudo systemctl enable sshd
+          sudo systemctl start sshd
+          sudo systemctl status sshd
+      - name: reserve machine from testing farm
+        run: |
+          ssh-keygen -t rsa -N '' -f id_rsa -q 
+          export TESTING_FARM_API_TOKEN=059e594d-12da-4ec6-951f-9069e99cf2c8
```

Comment:
My feeling is that this API key should be a secret? https://docs.github.com/en/actions/security-for-github-actions/security-guides/using-secrets-in-github-actions

Reply:
@cfergeau yes, the self-hosted runner failed to get the secret. I'm still working on it. This is temporary to verify the Testing Farm machine can be successfully reserved.

Reply:
@lilyLuLiu In this case `059e594d-12da-4ec6-951f-9069e99cf2c8` one is testing key and doesn't provide any access to the user?

Reply:
@praveenkumar The token is required to reserve machine from Testing Farm. If the token stored in Github secret, action triggered by PR from forked repo can't access the secret.  Then the PR can't reserve machine successfully.

Reply:
> If the token stored in Github secret, action triggered by PR from forked repo can't access the secret. 

I think you need to use workflow like https://github.com/crc-org/crc/blob/main/.github/workflows/windows-qe-tpl.yml one where you can consume the secret from the PR also. 
- https://securitylab.github.com/resources/github-actions-preventing-pwn-requests/ have more details